api/v1alpha1/tcproute_types.go

```diff
@@ -16,44 +16,51 @@ limitations under the License.
 package v1alpha1
 
 import (
+	core "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
-// TcpRouteSpec defines the desired state of TcpRoute
-type TcpRouteSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+// TCPRouteSpec defines the desired state of the TCPRoute.
+type TCPRouteSpec struct {
+	// Hosts is a list of Host definitions.
+	Hosts []TCPRouteHost `json:"hosts,omitempty"`
```

Comment:
I think it would be better to call this `Rules` instead. 
Also, `s/TCPRouteHost/TCPRouteRule`.

Reply:
I wanted to keep the structure of `TCPRoute` similar to the structure of `HTTPRoute` where possible.  Should we make the change you are suggesting to `HTTPRoute` as well?

Reply:
HTTPRoute is up in the air right now because we are still working through the relationship of Gateway, HTTPRoute and `host`.

THe current api is:
```yaml
hosts:
- host: "foo.com"
  destinations # will be in future
- host: "bar.com"
SNIP
```

`hosts` seems not the best name here since it is not an array of `hosts` but something more.